template/src/util/api.ts

```diff
@@ -157,6 +161,77 @@ export class API {
         )
     }
 
+    /**
+     * Determines via introspection if the GraphQL API has local code intelligence available
+     *
+     * TODO(chrismwendt) - Remove this when we no longer need to support versions without local code
+     * intelligence
+     */
+    public hasLocalCodeIntelField = once(async () => {
+        const introspectionQuery = gql`
+            query LocalCodeIntelIntrospectionQuery {
+                __type(name: "GitBlob") {
+                    fields {
+                        name
+                    }
+                }
+            }
+        `
+
+        interface IntrospectionResponse {
+            __type: { fields: { name: string }[] }
+        }
+
+        return (await queryGraphQL<IntrospectionResponse>(introspectionQuery)).__type.fields.some(
+            field => field.name === 'localCodeIntel'
+        )
+    })
+
+    public fetchPayload = cache(
```

Comment:
Can we name this more specifically?

Reply:
Yup, renamed to `fetchLocalCodeIntelPayload`.